### PR TITLE
Add v0.10.0 release notes

### PR DESCRIPTION
**Pull request by jedevc**

### CODE REVIEW

File: .changes/v0.10.0.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+## v0.10.0 - 2024-02-27
+
+
+### Added
+- New [Dagger functions and modules](https://docs.dagger.io/developer-guide/overview/482011/architecture)
```
Comment (gerhard) [inline .changes/v0.10.0.md:5]: To me, this seems like a better link: https://deploy-preview-6750--devel-docs-dagger-io.netlify.app/guides/457482/create-app-ci-module

This is what the PR is currently linking to: https://deploy-preview-6750--devel-docs-dagger-io.netlify.app/developer-guide/overview/482011/architecture

WDYT @vikram-dagger?

**Review by gerhard — COMMENTED** [review]
I really like this changelog!

I am unsure about the docs page that we are linking to. WDYT about my proposed alternative?

File: .changes/v0.10.0.md (hunk 1 above)
Comment (jedevc) [reply .changes/v0.10.0.md:5]: Don't think the guide is quite the right one - to quote:

> This guide assumes that:
> - You know the basics of calling and writing Dagger Modules

Maybe we should link to the Quickstart instead? https://deploy-preview-6750--devel-docs-dagger-io.netlify.app/quickstart

Comment (jedevc) [reply .changes/v0.10.0.md]: Updated to point to the quickstart instead - we can later update this if we need to, and modify the github release description.

Comment (gerhard) [reply .changes/v0.10.0.md]: OK!

**Review by gerhard — APPROVED** [review]
👍 🚀

**Pull request merged**